internal/eval/evalers.go

```diff
@@ -804,6 +805,73 @@ func (n *hasEval) Eval(env Env) (types.Value, error) {
 	return types.Boolean(ok), nil
 }
 
+// getTagEval
+type getTagEval struct {
+	lhs, rhs Evaler
+}
+
+func newGetTagEval(object, tag Evaler) *getTagEval {
+	return &getTagEval{lhs: object, rhs: tag}
+}
+
+func (n *getTagEval) Eval(env Env) (types.Value, error) {
+	eid, err := evalEntity(n.lhs, env)
+	if err != nil {
+		return zeroValue(), err
+	}
+
+	var unspecified types.EntityUID
+	if eid == unspecified {
+		return zeroValue(), fmt.Errorf("cannot access tag `%s` of %w", n.rhs, errUnspecifiedEntity)
+	}
```

Comment:
I think I would call this a "zero" ID rather than an "unspecified" ID.

That being said, it appears that entity type names cannot be empty, but the id part of the string can be. The Cedar text unmarshaler wouldn't allow this entity to exist, but I think the JSON unmarshaler and `NewEntityUID()` would.

I think we should actually file a bug on the JSON unmarshaler for allowing empty type names. We should also really enforce the type name being non-empty in `NewEntityUID()`, but adding an error return type to that function would be a breaking change. Maybe we could panic?

All that is to say, I'm not sure how important it really is to consider this case in the evaler.

Reply:
Yes, I think that makes sense, thanks for expounding on it!

I've renamed this to `zero` here: https://github.com/cedar-policy/cedar-go/pull/77/commits/9074d5b00f44be4346c13ac8b7270a09b06c541a.